### ISSUE TITLE
Can't get incoming texts

### ISSUE THREAD

**Issue by K-RAD** [Skylar-Tech/node-red-contrib-matrix-chat#93]
I connected it to my own beeper account and I have a weird issue, where I can send messages, I can get incoming emojis, but not the incoming text messages.

**Comment by wuast94** [comment]
can you rovide more information on this?

**Comment by skylord123** [comment]
Yeah I can't replicate this.

What version of Node-RED are you running?
What version of NodeJS is Node-RED running on?
Is the server you are connecting to self hosted or a public one? If self hosted what server type and version is it?

I upgraded my server to the latest matrix release to see if I could replicate this and I could not.

**Comment by K-RAD** [comment]
I'm running node-red 3.0.2 in docker (on a synology nas)
nodejs 16.20.1
node-red-contrib-matrix-chat 0.7.0

Could it be something related to encryption? I can see that the emojis are not encrypted

**Comment by skylord123** [comment]
> Could it be something related to encryption? I can see that the reactions are not encrypted

If it is an issue with encryption you will see it in your logs for Node-RED. Check there and see if you have anything after receiving the message.

**Comment by K-RAD** [comment]
> If it is an issue with encryption you will see it in your logs for Node-RED. Check there and see if you have anything after receiving the message.

You're right, there's a lot of [m.bad.encrypted] logs, depending on which chat platform they originate.

- Requested session ID not found on this device
- The sender's device has not sent us the keys for this message
- The sender has disabled encrypting to unverified devices

**Comment by skylord123** [comment]
> no.. only reactions

I believe this is because reactions aren't encrypted currently. There is no real good way around this. Once a client breaks because it was used elsewhere and not strictly in node-red you will start having encryption keys stored on the other client that node-red has no clue about. This is why it's important to strictly only use the user with Node-RED if you are going the encryption route.

#87 is for adding the ability to verify devices and do key sharing. I've looked into it a few times and have not had much success with it. The matrix-js-sdk library doesn't have the best built in support for handling it so we have to do it manually and I do not have enough expertise in the matrix spec to make it happen.

There is another solution by using pantalaimon as a proxy between Node-RED and your matrix server. It will handle all the encryption for you.

Would love for someone to step up and help us with getting device validation and key sharing working.

**Comment by skylord123** [comment]
@K-RAD — v1.0.0 should resolve this. The `m.bad.encrypted` errors you saw (*"sender has disabled encrypting to unverified devices"* and *"sender's device has not sent us the keys for this message"*) are both symptoms of the bot's session being unverified — other clients see the bot as untrusted and refuse to share the megolm room keys with it, so its incoming messages can't be decrypted. Cross-signing and verifying the bot fixes both.

v1.0.0 has the missing pieces — `matrix-js-sdk` upgraded to 41.5.0 with the Rust crypto stack (#122), cross-signing, secure backup, and device verification (#87/#124).

Two ways to verify, depending on whether you want a flow or not:

- **From the server config node** — the new **Pending verification requests** button lists incoming requests and walks you through the SAS (emoji) check interactively, no flow required. The same dialog also has **Set up secure backup & cross-signing** (run this once so the bot's own session shows as verified to other clients — this is the part that unblocks the bridged platforms refusing to send to you) and **Manage sessions**.
- **From a flow** — the new `matrix-verification` (event source) and `matrix-verification-action` (request / accept / start SAS / confirm / cancel) nodes.

A v1.0.0 beta is out — please give it a try:

```
npm install node-red-contrib-matrix-chat@beta
```

Full changelog and tracking PR: #142 · release notes: https://github.com/Skylar-Tech/node-red-contrib-matrix-chat/releases/tag/v1.0.0-beta.1
